src/control/server/storage_scm_test.go

```diff
@@ -303,15 +352,37 @@ func TestFormatScm(t *testing.T) {
 			config := newMockStorageConfig(tt.mountRet, tt.unmountRet,
 				tt.mkdirRet, tt.removeRet, tt.mount, tt.class, tt.devs,
 				tt.size, storage.BdevClassNvme, []string{}, false, false)
+
+			mockPrep := newMockPrepScm()
+			if tt.prepNoNdctl {
+				err := errors.New(msgNdctlNotFound)
+				mockPrep = &mockPrepScm{
+					prepRet:          err,
+					resetRet:         err,
+					getStateRet:      err,
+					getNamespacesRet: err,
+				}
+			}
+
 			ss := newMockScmStorage(log, config.ext, nil, []DeviceDiscovery{},
-				false, newMockPrepScm())
+				false, mockPrep)
 			ss.formatted = tt.formatted
 
 			results := ScmMountResults{}
 
 			if tt.inited {
+				// Discovery is run in SCS.Setup() and is not
+				// fatal, continue with expected errors to
+				// format as in normal program execution.
 				if err := ss.Discover(); err != nil {
-					t.Fatal(err)
+					if tt.prepNoNdctl {
+						ExpectError(t, err,
+							msgIpmctlDiscoverFail+": "+msgNdctlNotFound,
+							tt.desc)
+					} else {
+						// unexpected failure
+						t.Fatal(tt.desc + ": " + err.Error())
+					}
```

Comment:
Just a thought, but the growing conditional logic in the body of this loop is starting to feel a bit complex. Wondering if instead of a boolean that the loop logic takes care of, mockProp and expectedError could be members of the test case struct. I think that would simplify the logic.

Reply:
sure